packages/graphql-codegen-typescript-operations/src/ts-pre-resolve-types-processor.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  getBaseType,
+  removeNonNullWrapper,
+} from "@graphql-codegen/plugin-helpers";
+import {
+  PreResolveTypesProcessor as CodegenPreResolveTypesProcessor,
+  PrimitiveField,
+  ProcessResult,
+  SelectionSetProcessorConfig,
+} from "@graphql-codegen/visitor-plugin-common";
+import {
+  GraphQLInterfaceType,
+  GraphQLObjectType,
+  isEnumType,
+  isNonNullType,
+} from "graphql";
+
+interface CustomSelectionSetProcessorConfig
+  extends SelectionSetProcessorConfig {
+  inlineCommonTypes?: boolean;
+}
+export class PreResolveTypesProcessor extends CodegenPreResolveTypesProcessor {
+  private inlineCommonTypes: boolean;
+
+  constructor(config: CustomSelectionSetProcessorConfig) {
+    super(config);
+    this.inlineCommonTypes = config.inlineCommonTypes ?? false;
+  }
+  transformPrimitiveFields(
+    schemaType: GraphQLObjectType | GraphQLInterfaceType,
+    fields: PrimitiveField[],
+  ): ProcessResult {
+    if (fields.length === 0) {
+      return [];
+    }
+
+    return fields.map((field) => {
+      const fieldObj = schemaType.getFields()[field.fieldName];
+
+      const baseType = getBaseType(fieldObj.type);
+      let typeToUse = baseType.name;
+
+      const useInnerType = field.isConditional && isNonNullType(fieldObj.type);
+      const innerType = useInnerType
+        ? removeNonNullWrapper(fieldObj.type)
+        : undefined;
```

Comment:
So if `fieldObj.type` is something like `[[[MyEnum!]!]!]!` - this will still work properly? Nested wrapping types are always tricky, 
let's add test cases for those (unless we already have those)?

E.g. 
1. All nullable: `[[[MyEnum]]]`
2. All non-nullable: `[[[MyEnum!]!]!]!`
3. A mix of those: `[[[MyEnum!]]!]`